packages/@icon-magic/generate/src/plugins/svg-light-dark.ts

```diff
@@ -31,163 +30,85 @@ export const svgLightDark: GeneratePlugin = {
 
         // write the parent svg with light/dark children svg's to the output directory
         await fs.ensureDir(outputPath);
         await fs.writeFile(
-          path.format({dir: outputPath, name: `${imageSet}-mixed`, ext: '.svg'}),
-          mixedAsset.data,
+          path.format({dir: outputPath, name: `${imageset}-mixed`, ext: '.svg'}),
+          serializeToString(wrapperEl),
           {
             encoding: 'utf8'
           }
         );
 
         // Final new Mixed Flavor
         const mixedFlavor: Flavor = new Flavor(icon.iconPath, {
-          name: `${imageSet}-mixed`,
-          path: `./${imageSet}-mixed.svg`,
-          colorScheme: 'mixed'
+          name: `${imageset}-mixed`,
+          path: `./${imageset}-mixed.svg`,
+          colorScheme: 'mixed',
+          imageset: flavor.imageset
         });
 
+        mixedFlavor.types.set('svg', mixedFlavor);
+
         // Add new mixed flavor to icon.flavors. It is then added to resulting iconrc.json file.
         icon.flavors.set(
-          `${imageSet}-mixed`,
+          `${imageset}-mixed`,
           mixedFlavor
         );
         return mixedFlavor;
       }
     }
+    // do nothing and return the original flavor
     return flavor;
   }
 };
-
-// TODO: lightAsset type would be `OptimizedSvg` from Svgo interface, but it does not accept this type, looks like OptimizedSvg is not exported from Svgo.
-
-/**
- * Takes two OptimizedSvg's from Svgo interface, adds them as children to a
- * parent SVG Dom element and returns the Dom element
- * @param lightAsset `OptimizedSvg` from Svgo interface
- * @param darkAsset `OptimizedSvg` from Svgo interface
- * @returns SVGSVGElement Dom elment with the two parameters as children.
- */
-async function generateMixedSvgParent(lightAsset: any, darkAsset: any): Promise<SVGSVGElement> {
-  // Create SVG PARENT
-  const DOM = new DOMImplementation();
-  const doctype = DOM.createDocumentType(
-    'svg',
-    '-//W3C//DTD SVG 1.1//EN',
-    'http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd'
-  );
-  const SVG_NS = 'http://www.w3.org/2000/svg';
-
-  // Create an SVG Document and set its doctype
-  const DOCUMENT = DOM.createDocument(SVG_NS, 'svg', doctype);
-
-  // Create SVG element
-  const svgEl = DOCUMENT.createElementNS(SVG_NS, 'svg');
-  svgEl.setAttribute('xmlns', 'http://www.w3.org/2000/svg');
-
-  // Add <svg> element to SVG Document
-  DOCUMENT.appendChild(svgEl);
-
-  const svgLight = lightAsset.data;
-  const svgDark = darkAsset.data;
-
-  // Parse XML from a string into a DOM Document.
-  const doc = new DOMParser();
-  const xmlLight = doc.parseFromString(svgLight, 'image/svg+xml');
-  const xmlDark = doc.parseFromString(svgDark, 'image/svg+xml');
-
-  svgEl.setAttribute('width', xmlLight.documentElement.getAttribute('width') || '');
-  svgEl.setAttribute('height', xmlLight.documentElement.getAttribute('height') || '');
-  svgEl.setAttribute('viewBox', xmlLight.documentElement.getAttribute('viewBox') || '');
-
```

Comment:
So is wrapper SVG still getting these, I wasn't totally sure. Thx

Reply:
Yes, the wrapper SVG is a clone of the outer SVG of the light asset. Refer to line 64. This means that if the light asset has the above attributes, then the mixed asset will have them too. This assumes that light and dark have the same attributes (which usually should be the case since they're going through svg-generate with the same attributes)